test/js/bun/test/expect-extend.test.js

```diff
@@ -0,0 +1,213 @@
+  expect.extend({
+    _toBeOne(_expected) {
+      expect(this.equals).toBeFunction();
+      return { message: () => "", pass: !!this.equals(1, 1) };
+    },
+  });
+
+  expect(() => expect("test")._toBeOne()).not.toThrow();
+});
+
+it("defines asymmetric unary matchers", () => {
+  expect(() => expect({ value: 2 }).toEqual({ value: expect._toBeDivisibleBy(2) })).not.toThrow();
+  expect(() => expect({ value: 3 }).toEqual({ value: expect._toBeDivisibleBy(2) }))._toThrowErrorMatchingSnapshot();
+  expect(() => expect({ value: 3 }).toEqual({ value: expect._toBeDivisibleBy(2) })).toThrow();
+});
+
+it("defines asymmetric unary matchers that can be prefixed by not", () => {
+  expect(() => expect({ value: 2 }).toEqual({ value: expect.not._toBeDivisibleBy(2) }))._toThrowErrorMatchingSnapshot();
+  expect(() => expect({ value: 3 }).toEqual({ value: expect.not._toBeDivisibleBy(2) })).not.toThrow();
+});
+
+it("defines asymmetric variadic matchers", () => {
+  expect(() => expect({ value: 2 }).toEqual({ value: expect._toBeWithinRange(1, 3) })).not.toThrow();
+  expect(() => expect({ value: 3 }).toEqual({ value: expect._toBeWithinRange(4, 11) }))._toThrowErrorMatchingSnapshot();
+});
+
+it("defines asymmetric variadic matchers that can be prefixed by not", () => {
+  expect(() =>
+    expect({ value: 2 }).toEqual({
+      value: expect.not._toBeWithinRange(1, 3),
+    }),
+  )._toThrowErrorMatchingSnapshot();
+  expect(() =>
+    expect({ value: 3 }).toEqual({
+      value: expect.not._toBeWithinRange(5, 7),
+    }),
+  ).not.toThrow();
+});
+
+it("prints the Symbol into the error message", () => {
+  const foo = Symbol("foo");
+  const bar = Symbol("bar");
+
+  expect(() =>
+    expect({ a: foo }).toEqual({
+      a: expect._toBeSymbol(bar),
+    }),
+  )._toThrowErrorMatchingSnapshot();
+});
+
+it("allows overriding existing extension", () => {
+  expect.extend({
+    _toAllowOverridingExistingMatcher(_expected) {
+      return { message: () => "", pass: _expected === "bar" };
+    },
+  });
+
+  expect("foo").not._toAllowOverridingExistingMatcher();
+
+  expect.extend({
+    _toAllowOverridingExistingMatcher(_expected) {
+      return { message: () => "", pass: _expected === "foo" };
+    },
+  });
+
+  expect("foo")._toAllowOverridingExistingMatcher();
+});
+
+it("throws descriptive errors for invalid matchers", () => {
+  expect(() =>
+    expect.extend({
+      // @ts-expect-error
+      default: undefined,
+    }),
+  ).toThrow(/*'expect.extend: `default` is not a valid matcher. Must be a function, is "undefined"'*/);
+  expect(() =>
+    expect.extend({
+      // @ts-expect-error
+      default: undefined,
+    }),
+  ).toThrow(/*'expect.extend: `default` is not a valid matcher. Must be a function, is "undefined"'*/);
+  expect(() =>
+    expect.extend({
+      // @ts-expect-error
+      default: 42,
+    }),
+  ).toThrow(/*'expect.extend: `default` is not a valid matcher. Must be a function, is "number"'*/);
+  expect(() =>
```

Comment:
were these meant to be commented? they still pass uncommented

Reply:
forgot about them, thanks